src/main/scala/nl/knaw/dans/easy/multiDeposit/Main.scala

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (C) 2015-2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.multiDeposit
+
+import java.io.File
+
+import nl.knaw.dans.easy.multiDeposit.actions._
+import nl.knaw.dans.easy.multiDeposit.{CommandLineOptions => cmd, MultiDepositParser => parser}
+import org.slf4j.LoggerFactory
+import rx.lang.scala.{Observable, ObservableExtensions}
+import rx.schedulers.Schedulers
+
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+object Main {
+
+  implicit val log = LoggerFactory.getLogger(getClass)
+
+  def main(args: Array[String]) {
+    log.debug("Starting application.")
+    implicit val settings = cmd.parse(args)
+    getActionsStream
+      .doOnError(e => log.error(e.getMessage, e))
+      .doOnCompleted { log.info("Finished successfully!") }
+      .subscribe
+
+    Schedulers.shutdown()
+  }
+
+  def getActionsStream(implicit settings: Settings): Observable[Action] = {
+    parser.parse(new File(settings.multidepositDir, cmd.mdInstructionsFileName))
+      .getActions
+      .checkActionPreconditions
+      .runActions
+  }
+
+  // Extension methods are used here to do composition
+  // (see http://reactivex.io/rxscala/comparison.html)
+  implicit class ActionExtensionMethods(val actions: Observable[Action]) extends AnyVal {
+    def checkActionPreconditions = Main.checkActionPreconditions(actions)
+
+    def runActions = Main.runActions(actions)
+  }
+
+  implicit class DatasetsExtensionMethods(val datasets: Observable[Datasets]) extends AnyVal {
+    def getActions(implicit s: Settings) = datasets.flatMap(Main.getActions)
+  }
+
+  def getActions(dss: Datasets)(implicit s: Settings): Observable[Action] = {
+    log.info("Compiling list of actions to perform ...")
+    val tryActions = dss.flatMap(getDatasetActions).toList :+
+      Success(CreateSpringfieldAction(-1, dss)) // SpringfieldAction runs on multiple rows, so -1 here
+
```

Comment:
I am normally all for it to have as few comments as possible, but I think the next part needs some clarification.